contrib/checkstyle/src/com/etsy/sbt/checkstyle/CheckstyleConfigLocation.scala

```diff
@@ -0,0 +1,43 @@
+package com.etsy.sbt.checkstyle
+
+import os.Path
+
+import scala.io.Source
+import upickle.default._
+
+/**
+ * Represents a Checkstyle XML configuration located locally, on the class path or remotely at a URL
+ *
+ * @author Joseph Earl
+ */
+sealed trait CheckstyleConfigLocation {
+  def read(resources: Seq[Path]): String
+}
```

Comment:
Could this trait hierarchy be simplified? `Classpath` seems strange, since it's loading stuff from the Mill classloader, which won't contain any of the application classpath. 

`URL` seems redundant, since if we ask for a file, Mill can convert that into a URL by doing `override def foo = T{ requests.get("...") }`, so I don't think we need to have a special case for it in the configuration ADT.

Given that, it might be best to just make the config a `PathRef`, and led anyone who wants to fetch that file from unusual locations just do an `override` the task to call `requests.get` or `os.read(os.resource / ...)` themselves

Reply:
`Classpath` uses the application's dependencies (if you comment out the dependency in the `checkstyleClasspathConfig` test it will fail). I think that's an important use case (use a config that's published in some organisational jar). Happy to cut `URL` if you still think that's worthwhile given that we do need the ADT?